lib/runtime/src/component.rs

```diff
@@ -268,9 +269,11 @@ impl Component {
 
     /// Add Prometheus metrics for this component's service stats.
     ///
-    /// Uses a channel to synchronize with the spawned async task, ensuring
-    /// metrics are updated before the callback returns.
-    pub fn add_metrics_callback(&self) -> Result<()> {
+    /// Starts a background task that scrapes stats every 5 seconds and updates metrics.
+    /// The callback will simply use the latest scraped data.
+    pub fn start_scraping_metrics(&self) -> Result<()> {
+        const MAX_DELAY: std::time::Duration = std::time::Duration::from_millis(4700);
```

Comment:
Can you describe or give a name/const to the 4700 here? I'm guessing it's something like 5000ms (5 seconds) minus the 300ms later on in the code - but this could be made clearer with some consts